apps/epic-web/src/inngest/functions/ai-email/write-an-email.ts

```diff
@@ -0,0 +1,264 @@
+import {inngest} from 'inngest/inngest.server'
+import {
+  EMAIL_WRITING_REQUEST_COMPLETED_EVENT,
+  EMAIL_WRITING_REQUESTED_EVENT,
+} from 'inngest/events'
+import OpenAI from 'openai'
+
+const openai = new OpenAI()
+
+export const writeAnEmail = inngest.createFunction(
```

Comment:
this is a chain of calls to gpt-4 that goes writer->editor->writer->editor with a final step that fixes/validates the json